packages/utils/src/collect.ts

```diff
@@ -1,82 +1,29 @@
-import {
-  GroupedObservable,
-  Observable,
-  OperatorFunction,
-  pipe,
-  Subscription,
-} from "rxjs"
-import {
-  startWith,
-  endWith,
-  ignoreElements,
-  map,
-  distinctUntilChanged,
-  skipWhile,
-} from "rxjs/operators"
+import { GroupedObservable, noop, Observable } from "rxjs"
 import { CollectorAction, collector } from "./internal-utils"
 
-const defaultFilter = pipe(ignoreElements(), startWith(true), endWith(false))
-
-export type CollectedObservable<K, V> = Observable<
-  Map<K, GroupedObservable<K, V>>
-> & {
-  get: (key: K) => Observable<V>
-}
+const enhancer = <K, V>(source: GroupedObservable<K, V>) =>
+  new Observable<
+    | { t: CollectorAction.Set; k: K; v: GroupedObservable<K, V> }
+    | { t: CollectorAction.Delete; k: K }
+  >((observer) => {
+    const message = {
+      t: CollectorAction.Set as const,
+      k: source.key,
+      v: source,
+    }
+    observer.next(message)
+    ;(message as any).t = CollectorAction.Delete
```

Comment:
smells like danger 😱

I see it currently doesn't hurt because the consumer of this enhancer reads the messages synchronously, but if it did asynchronously the initial message would get mutated from `Set` => `Delete`. You sure about this?

Edit - Thinking it better, yep it's not that important b/c it's an internal implementation - If it ever breaks the external behaviour, tests would catch that

Reply:
I realize that we no longer need the `CollectorAction` enum :partying_face: